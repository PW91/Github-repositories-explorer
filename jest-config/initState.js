export default {
  ui: {
    activeUserCard: null,
    lastSearchQuery: null,
    isSearchEmpty: null,
    isLoading: null
  },
  users: []
}