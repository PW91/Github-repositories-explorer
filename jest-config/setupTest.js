import Enzyme from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import 'regenerator-runtime/runtime.js'
import '~/i18n'

Enzyme.configure({
  adapter: new EnzymeAdapter(),
  disableLifecycleMethods: true
})
