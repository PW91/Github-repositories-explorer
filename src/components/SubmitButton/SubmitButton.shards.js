import styled from '@emotion/styled'

export const BaseButton = styled.button`
  text-transform: uppercase;
  font-size: ${({ theme }) => theme.fontSize.text.small};
  border-radius: ${({ theme }) => theme.borderRadius.main};
  cursor: pointer;
  user-select: none;
  border: none;
  padding: 1rem 2rem;
  transition: background-color .1s;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.button.accept.main};
  color: ${({ theme }) => theme.colors.text.contrast};

  &:focus {
    outline: none;
  }

  &:disabled {
    pointer-events: none;
    background-color: ${({ theme }) => theme.colors.button.accept.disabled};
  }

  &:hover {
    background-color: ${({ theme }) => theme.colors.button.accept.hover};
  }
`
