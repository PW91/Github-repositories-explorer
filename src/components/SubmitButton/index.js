export { default } from '~/components/SubmitButton/SubmitButtonView'
