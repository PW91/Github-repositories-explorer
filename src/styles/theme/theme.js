import * as colors from '~/styles/theme/colors'
import * as commons from '~/styles/theme/commons'

export default {
  colors,
  ...commons
}
