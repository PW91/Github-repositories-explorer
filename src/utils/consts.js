export const API_URL = 'https://api.github.com'

export const BREAKPOINTS = {
  SM: 480,
  MD: 768,
  LG: 992,
  XL: 1300,
}

export const SEARCH_RESULTS_LIMIT = 5

export const API_ACTIONS = {
  USERS_SEARCH: 'USERS_SEARCH',
  USER_REPOS: 'USER_REPOS'
}

export const STORE_ACTIONS = {
  SET_ACTIVE_USER_CARD: 'SET_ACTIVE_USER_CARD',
  SET_LAST_SEARCH_QUERY: 'SET_LAST_SEARCH_QUERY',
  SET_SEARCH_EMPTY_FLAG: 'SET_SEARCH_EMPTY_FLAG',
  SET_LOADING_FLAG: 'SET_LOADING_FLAG',
  SET_USERS: 'SET_USERS',
  SET_USER_REPOS: 'SET_USER_REPOS'
}
