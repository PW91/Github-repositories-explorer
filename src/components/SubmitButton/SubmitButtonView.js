import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'formik'
import { BaseButton } from '~/components/SubmitButton/SubmitButton.shards'

const SubmitButtonView = ({ children, formik }) => {
  const { isSubmitting, isValid } = formik
  return (
    <BaseButton
      data-test='SubmitButtonView'
      type='submit'
      disabled={!isValid || isSubmitting}
    >
      {children}
    </BaseButton>
  )
}

SubmitButtonView.propTypes = {
  children: PropTypes.string,
  formik: PropTypes.object
}

export default connect(SubmitButtonView)
