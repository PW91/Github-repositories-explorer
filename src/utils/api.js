import { format } from 'url'
import { toast } from 'react-toastify'
import { parseJSON } from '~/utils/helpers'
import { API_URL, API_ACTIONS } from '~/utils/consts'

const ajaxer = (apiAction, path, method = 'GET', data = null, query = null) => (
  new Promise((resolve, reject) => {
    const sendForm = data && data.constructor === FormData
    const requestOptions = {
      headers: !sendForm && { 'content-type': 'application/json' },
      method,
      ...!!data && { body: sendForm ? data : JSON.stringify(data) },
    }

    const fullPath = `${API_URL}${path}${query ? format({ query }) : ''}`

    fetch(fullPath, requestOptions)
      .then(res => res.status < 400 ? res.text() : { errorText: res.statusText, status: res.status })
      .then(res => {
        if (res.errorText) {
          const { errorText, status } = res
          toast.error(errorText)
          reject({ errorText, status })
        } else {
          const parsedJson = parseJSON(res)
          const mapped = responseFormatter(parsedJson, apiAction)
          resolve(mapped)
        }
      })
  })
)

export const responseFormatter = (data, apiAction) => {
  switch (apiAction) {
    case API_ACTIONS.USERS_SEARCH:
      return data.items.map(({
        login,
        id,
        avatar_url
      }) => ({
        login,
        id,
        avatarUrl: avatar_url
      }))
    case API_ACTIONS.USER_REPOS:
      return data.map(({
        id,
        owner,
        name,
        html_url,
        stargazers_count,
        description
      }) => ({
        id,
        owner: owner.id,
        name,
        htmlUrl: html_url,
        stars: stargazers_count,
        description
      }))
  }
}

export const manageUsersSearch = {
  get: (query) => ajaxer(API_ACTIONS.USERS_SEARCH, '/search/users', 'GET', null, query)
}

export const manageUserRepos = {
  get: (username) => ajaxer(API_ACTIONS.USER_REPOS, `/users/${username}/repos`)
}
