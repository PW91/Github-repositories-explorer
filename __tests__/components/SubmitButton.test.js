import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import serializer from 'jest-emotion'
import { ThemeProvider } from 'emotion-theming'
import { findByTestAtrr } from '>/jest-config/utils'
import theme from '~/styles/theme/theme'
import SubmitButton from '~/components/SubmitButton'

expect.addSnapshotSerializer(serializer)

describe('SubmitButton component', () => {

  it('It renders', () => {
    const wrapper =  shallow(<SubmitButton />).dive().dive()
    expect(findByTestAtrr(wrapper, 'SubmitButtonView').exists()).toBeTruthy()
  })

  it('Snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <SubmitButton />
      </ThemeProvider>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
