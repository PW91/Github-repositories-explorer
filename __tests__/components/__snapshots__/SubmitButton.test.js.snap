// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton component Snapshot 1`] = `
.emotion-0 {
  text-transform: uppercase;
  font-size: 1.4rem;
  border-radius: .2rem;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: none;
  padding: 1rem 2rem;
  -webkit-transition: background-color .1s;
  transition: background-color .1s;
  width: 100%;
  background-color: #66c300;
  color: #ffffff;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:disabled {
  pointer-events: none;
  background-color: #757575;
}

.emotion-0:hover {
  background-color: #1ba307;
}

<button
  className="emotion-0 emotion-1"
  data-test="SubmitButtonView"
  disabled={true}
  type="submit"
/>
`;
